File: packages/functions/ios/Plugin/Classes/Options/CallOptions.swift
import Foundation

@objc public class CallOptions: NSObject {
    private var data: Any?
    private var timeout: Int?

    init(data: Any?, timeout: Int?) {
        self.data = data
        self.timeout = timeout
    }

    func getData() -> Any? {
        return data
    }

    func getTimeout() -> Int? {
        return timeout
    }
}

File: packages/functions/android/src/main/java/app/capgo/capacitor/firebase/functions/CallOptions.java
package app.capgo.capacitor.firebase.functions.classes.options;

import androidx.annotation.Nullable;

public class CallOptions {

    @Nullable
    private Object data;

    @Nullable
    private Long timeout;

    public CallOptions(@Nullable Object data, @Nullable Long timeout) {
        this.data = data;
        this.timeout = timeout;
    }

    @Nullable
    public Object getData() {
        return data;
    }

    @Nullable
    public Long getTimeout() {
        return timeout;
    }
}
